homeassistant/components/shelly/__init__.py

```diff
@@ -48,9 +54,13 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     except (asyncio.TimeoutError, OSError) as err:
         raise ConfigEntryNotReady from err
 
-    wrapper = hass.data[DOMAIN][entry.entry_id] = ShellyDeviceWrapper(
-        hass, entry, device
-    )
+    undo_listener = entry.add_update_listener(update_listener)
```

Comment:
We don't need this. The default is to reload after options are updated.

Reply:
Actually that's not the case, I thought so too and tested it, but it doesn't work without it (I need to restart HA to see config changes).